### PR TITLE
merge socket constants

### PR DESCRIPTION
**Pull request by ndusart**
> Refactor the constant declarations such that all constants are only declared once with a #[cfg] that only enables the constant on the correct platforms.

#637

### CODE REVIEW

File: src/sys/socket/consts.rs [hunk 1]
```diff
@@ -1,75 +1,159 @@
...
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub const SOL_IPV6: c_int = libc::SOL_IPV6;
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub const SOL_NETLINK: c_int = libc::SOL_NETLINK;
     pub const IPPROTO_IP: c_int = libc::IPPROTO_IP;
     pub const IPPROTO_IPV6: c_int = libc::IPPROTO_IPV6;
     pub const IPPROTO_TCP: c_int = libc::IPPROTO_TCP;
+    #[cfg(any(target_os = "linux", target_os = "android", target_os = "dragonfly"))]
     pub const IPPROTO_UDP: c_int = SOL_UDP;
+    #[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "ios", target_os = "openbsd", target_os = "netbsd"))]
```
Comment (Susurrus) [inline src/sys/socket/consts.rs]: This line's too long. Please wrap lines to 100 characters. In this case, I like aligning all the `target_os` sections vertically. This applies to many other lines here, so please fix it for all of them.

Comment (ndusart) [inline src/sys/socket/consts.rs]: Should I split `target_os`  in several lines for each case, or only when the line is longer than 100 characters ?

Comment (Susurrus) [inline src/sys/socket/consts.rs]: Only for long lines.

**Comment by ndusart** [comment]
Don't know why but buildbot keep failing at fetching the code..

**Comment by asomers** [comment]
I don't know why either, but buildbot is trying a different sequence of commands for this build than it uses for most PRs.  Compare https://alan.ci/buildbot/#/builders/5/builds/200/steps/0/logs/stdio (this PR) to https://alan.ci/buildbot/#/builders/5/builds/79/steps/0/logs/stdio (PR 624).  I wonder if it's confused because your branch name is master?  Its my only guess at this point.

**Comment by ndusart** [comment]
Buildbot actually made two tests for each of my commits:

For the last one:
- https://alan.ci/buildbot/#/builders/5/builds/199
- https://alan.ci/buildbot/#/builders/5/builds/200

The first one succeeded though.

It's probably the name of my branch, I should take the habit to avoid working directly on the master branch once I fork the repo.

Do I need to create a new PR with another branch ?

**Comment by Susurrus** [comment]
@ndusart Could you try filing another PR using a feature branch? Then we can check if that's indeed the issue.

**Comment by ndusart** [comment]
This seems to be effectively the issue ! Succeed now with the new PR.

**Comment by Susurrus** [comment]
That's annoying. @asomers Looks like another bug that should be filed with the upstream buildbot project.

**Comment by Susurrus** [comment]
Closing in favor of #647.

**Pull request closed without merging**